src/librustc_trans/back/rpath.rs

```diff
@@ -40,9 +40,11 @@ pub fn get_rpath_flags(config: &mut RPathConfig) -> Vec<String> {
     let rpaths = get_rpaths(config, &libs);
     flags.extend_from_slice(&rpaths_to_flags(&rpaths));
 
-    // Use DT_RUNPATH instead of DT_RPATH if available
-    if config.linker_is_gnu {
-        flags.push("-Wl,--enable-new-dtags".to_string());
+    if cfg!(target_os = "linux") {
```

Comment:
I don't think this will work correctly when cross-compiling (e.g. host Linux, target BSD, or the other way round)

Reply:
I guess in link.rs we need to carry it forward from sess.target.target or something (just like linker_is_gnu is handled). But since I can't compile anything, I'm unlikely to get that right.